app/Models/BeatmapSet.php

```diff
@@ -536,6 +536,11 @@ public function storeCover($target_filename, $source_path)
         $this->storage()->put($this->coverPath().$target_filename, file_get_contents($source_path));
     }
 
+    public function previewURL()
+    {
+        return 'http://b.ppy.sh/preview/'.$this->beatmapset_id.'.mp3';
```

Comment:
Would it not be better to use

    return '//b.ppy.sh/preview/'.$this->beatmapset_id.'.mp3';

As forcing the content to load as HTTP will throw a content security warning when browsing with HTTPS.